Fixes zotero/translators#3124 by adding translator for TinRead

franklindyer opened the pull request:
See discussion of `zotero/translators#3124`. Added `TinREAD.js` which acts as a basic translator for TinRead records using exportable MARCXML files. Could probably still use some tweaking...

dstillman commented on TinREAD.js:
```diff
@@ -0,0 +1,277 @@
+}
+
+async function scrape(doc, _url = doc.location.href) {
+	doc.getElementById("DirectLink").click();
```
We can't rely on direct interaction with the webpage. While that will work in some current environments, it won't work in all environments (e.g., translation-server) and generally won't be possible in browsers in the future. So for a case like this, you'd want to figure out what the click kicked off and replicate that in code.

franklindyer commented on TinREAD.js:
```diff
@@ -0,0 +1,277 @@
+		translator.setTranslator("edd87d07-9194-42f8-b2ad-997c4c7deefd");
+		translator.setString(result);
+		translator.setHandler("itemDone", function (obj, item) {
+			finalize(doc, item);
```
Indeed, I noticed that a lot of the translators have a function called `finalize` even though it doesn't appear to be part of the [documented interface](https://www.zotero.org/support/dev/translators/coding#web_translators). I wasn't able to discern what `finalize` is generally used for, so I put it there for now as a kind of placeholder. 

Maybe you can shed some light on this? Should I remove it entirely?

franklindyer commented:
@AbeJellinek Thanks for all the feedback! I've implemented just about all of your suggestions, but the search page test is giving me a bit of trouble. I added a test, but it's currently failing with this opaque error message every time:

```
15:01:37 Translation using Library Catalog (TinREAD) failed: 
         Error: DOMParser error: loading data into data store failed
         
         Error: DOMParser error: loading data into data store failed
             parseDOMXML@chrome://zotero/content/xpcom/translate/src/translation/translate.js:2771:10
             getXML@chrome://zotero/content/xpcom/translate/src/translation/translate.js:2883:14
             importObject/attachTo[key]</<@chrome://zotero/content/xpcom/translation/translate_firefox.js:451:20
             doImport/<@chrome://zotero/content/xpcom/translation/translate_firefox.js line 425 > eval:59:13
             loadTranslator/safeTranslator.getTranslatorObject/promise<@chrome://zotero/content/xpcom/translate/src/translation/translate.js:414:21
         From previous event:
             loadTranslator/safeTranslator.translate@chrome://zotero/content/xpcom/translate/src/translation/translate.js:381:12
             scrape@chrome://zotero/content/xpcom/translation/translate_firefox.js line 425 > eval:100:8
         url => https://opac.biblioteca.ase.ro/opac/search?q=wirtschaftstheorie&max=0&view=&sb=relevance&ob=asc&level=all&material_type=all&do_file_type=all&location=0
         downloadAssociatedFiles => true
         automaticSnapshots => true
```

Not sure what this means, but I'll continue trying to troubleshoot.

It may be that I don't understand the search page tests very well in general. The example given [in the docs here](https://www.zotero.org/support/dev/translators/testing) seems to suggest that a test case for a search page can consist of no more than a check that `"multiple"` was correctly returned as the detected type, but the desired functionality is to provide the user with a list of results from which to select, which is more than this kind of test case checks for. Is it normal to have this minimal of a test case for search pages?

franklindyer commented:
Because of the debacle with issue #3225 I decided to try rerunning the search page test in Zotero 7 and... it worked! Huzzah!
